freqtrade/optimize/backtesting.py

```diff
@@ -161,7 +161,15 @@ def backtest(self, args: Dict) -> DataFrame:
         for pair, pair_data in processed.items():
             pair_data['buy'], pair_data['sell'] = 0, 0  # cleanup from previous run
 
-            ticker_data = self.populate_sell_trend(self.populate_buy_trend(pair_data))[headers]
+            ticker_data = self.populate_sell_trend(
+                self.populate_buy_trend(pair_data))[headers].copy()
+
+            # to avoid using data from future, we buy/sell with signal from previous candle
+            ticker_data.loc[:, 'buy'] = ticker_data['buy'].shift(1)
+            ticker_data.loc[:, 'sell'] = ticker_data['sell'].shift(1)
```

Comment:
would it make sense to extract these to each own functions which take the shift argument as parameter. Just thinking out that will we have a situation where we would need to control more of the given tickers?